### ISSUE TITLE
comma without a zero

### ISSUE THREAD

**Issue by heykanspor** [sosauce/CuteCalc#68]
hi, when i not type a zero before comma it gives an error. i tried it with the latest version.

**Comment by sosauce** [comment]
Hello, I need to see that with the guy who wrote the maths library 😁

**Comment by heykanspor** [comment]
hi, the another problem.

![Screenshot_20240816-233340](https://github.com/user-attachments/assets/04c5c71c-cbe1-4ae2-b90f-77b3b8c9df8b)
![Screenshot_20240816-233333](https://github.com/user-attachments/assets/da64daf5-17d1-4471-9f32-3ff1ff54cf56)

**Comment by sosauce** [comment]
This is normal, see issues #66 for more explanation

**Comment by notKamui** [comment]
> hi, when i not type a zero before comma it gives an error. i tried it with the latest version.

@heykanspor @sosauce `.5` is in no way a standardised notation for floating point numbers. Please write `0.5`. I will NOT support non standard numerical notation.

@sosauce You could pretreat the input string: if the character before the dot/comma is not a digit, you could insert a `0` there.

TLDR: this is basically not an issue.

**Comment by sosauce** [comment]
Yeah I was expecting you to say that, I wouldn't have bothered to ask anyways
